### PR TITLE
Add unstable log allocation config resource

### PR DESCRIPTION
**Pull request by lwatiker**
Introduces a new TF resource for defining log allocation configs for log budgets and adds support for logs dataset.

### CODE REVIEW

File: chronosphere/tfschema/log_allocation_config.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"allocation": LogAllocationSchema,
+			},
+			SchemaVersion: 1,
+		},
+		Required: true,
+		MaxItems: 1,
+	},
+	"dataset_allocation": {
```
Comment (lwatiker) [inline chronosphere/tfschema/log_allocation_config.go:19]: Made this singular to align with resource pools updated [schema](https://github.com/chronosphereio/terraform-provider-chronosphere/blob/2c97566d1fcb0c66c2840c4e9bb4890aa8efc4fe/chronosphere/tfschema/resource_pools_config.go#L42) but happy to make plural if thats preferred.

File: chronosphere/resource_log_allocation_config.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+		},
+	}
+}
+
+// LogAllocationConfigDryRunCount tracks how many times dry run is run during validation for testing.
+var LogAllocationConfigDryRunCount atomic.Int64
+
+type LogAllocationConfigConverter struct{}
+
+// -----
```
Comment (aschepis) [inline chronosphere/resource_log_allocation_config.go:56]: this is a tiny tiny optional nit. remove `-----`

File: chronosphere/tfschema/log_allocation_config.go (hunk 1 above)
Comment (aschepis) [reply chronosphere/tfschema/log_allocation_config.go:19]: i think that seems reasonable

Comment (aschepis) [reply chronosphere/tfschema/log_allocation_config.go:19]: does it clash with the naming schema of anything else?

**Review by aschepis — APPROVED** [review]

File: chronosphere/tfschema/log_allocation_config.go (hunk 1 above)
Comment (lwatiker) [reply chronosphere/tfschema/log_allocation_config.go:19]: We intentionally went with a naming scheme similar to resource pools. But don't think it should be a technical issue right?

**Pull request merged**